src/REL/d/a/e/d_a_e_sm.cpp

```diff
@@ -13,33 +90,1372 @@ STATE_DEFINE(dAcEsm_c, Absorption);
+                    if (lookRadius < 80.f) {
+                        lookRadius = 80.f;
+                    }
+
+                    dAcBomb_c *bomb = getBombWithinRadius(lookRadius);
+
+                    if (bomb != nullptr && std::abs(bomb->GetPosition().y - mHomePos1.y) < 0.7f * lookRadius) {
+                        if (bomb->mActorCarryInfo.tryAttachWithRef(bomb, this, &mBombRef, 1, false)) {
+                            mTimer_0xBAE = 160;
+                            playSound(SE_ESm_BRING_IN);
+                            clearActorProperty(1);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    if (mSph.ChkTgElectric()) {
+        fn_187_4B50();
+    }
+    s8 var = 0;
+    f32 radius = 150.f + var;
+    nw4r::g3d::ResNode node = mMdl.getModel().getResMdl().GetResNode("sm");
+
+    mMtx_c nodeMtx;
+    mMdl.getModel().getNodeWorldMtx(node.GetID(), nodeMtx);
+    mVec3_c center(0.f, 0.f, 0.f);
+    center.y = 90.f;
+
+    MTXMultVec(nodeMtx, center, mHomePos1);
+
+    position.CopyTo(poscopy3);
+    position.CopyTo(poscopy2);
+    poscopy2.y += 130.f * mScaleTarget.y;
+    poscopy3.y += 260.f * mScaleTarget.y;
+
+    mSph.SetC(mHomePos1);
+    if (mSph.ChkTgElectric()) {
+        playSoundEffect1(SE_ESm_ELEC_LV);
+    }
+
+    radius *= mScaleTarget.x;
+    if (radius < 43.f) {
+        radius = 43.f;
+    }
+    mSph.SetR(radius);
+    dCcS::GetInstance()->Set(&mSph);
+
+    f32 speedTarget = forwardSpeed;
+    speedTarget -= field_0xB80;
+    if (speedTarget < 0.f) {
+        speedTarget = 0.f;
+    }
+    sLib::addCalcScaledDiff(&forwardSpeed, speedTarget, 0.3f, 1.f);
+    sLib::addCalcScaled(&field_0xB80, 0.3f, 1.f);
+
+    if (field_0xBCB == 0) {
+        playSoundEffect1(SE_ESm_MOVE);
+    } else {
+        playSoundEffect1(SE_ESm_MOVE_HI);
+    }
+
+    if (field_0xBC8 == 0 && !mStateMgr.isState(StateID_Absorption)) {
+        fn_80030c20(3, 700.f, 50.f, -200.f, 200.f);
+    }
+
+    if (field_0xBBF != 3 && mScaleTarget.z != mScale.z) {
+        cLib::addCalcPos2(&mScale, mScaleTarget, 0.3f, 0.3f);
+    }
+
+    return SUCCEEDED;
+}
+
+int dAcEsm_c::draw() {
+    if (fn_187_5AC0()) {
+        return SUCCEEDED;
+    }
+    if (dAcPy_c::GetLink()->checkActionFlagsCont(0x400000)) {
+        return SUCCEEDED;
+    }
+    if (mScale.squareMagXZ() < 0.0004f) {
+        return SUCCEEDED;
+    }
+    if (shift8_0xFF != 0xFF && !SceneflagManager::sInstance->checkBoolFlag(roomid, shift8_0xFF)) {
+        return SUCCEEDED;
+    }
+
+    if (field_0xBBF == 3) {
+        return SUCCEEDED;
+    }
+
+    drawModelType1(&mMdl.getModel());
+
+    if (mHealth == 0) {
+        return SUCCEEDED;
+    }
+
+    // Wtf is this ordering...
+    s8 var = 0;
+    f32 y = 50.f;
+    f32 x = 0.f;
+    f32 z = 0.f;
+    mQuat_c q(x, y, z, mScale.x * (var + 240.f));
+    drawShadow(mShadowCircle, nullptr, mWorldMtx, &q, -1, -1, -1, -1, -1, position.y - mObjAcch.GetGroundH());
```

Comment:
The quat ctors are really annoying. Maybe we should collect problematic cases in an issue somewhere

Reply:
adding `mQuat_c(const mVec3_c &v, f32 w) : EGG::Quatf(w, v) {}` fixes this